x-pack/solutions/security/plugins/security_solution/common/constants.ts

```diff
@@ -572,3 +572,151 @@ export const PROMOTION_RULE_TAGS = [
   'Promotion', // This is the legacy tag for promotion rules and can be safely removed once promotion rules go live
   'Promotion: External Alerts',
 ];
+
+/**
+ * Essential fields to return for security alerts to reduce context window usage.
+ * These fields contain the most relevant information for security analysis.
+ */
+export const ESSENTIAL_ALERT_FIELDS: string[] = [
+  '_id',
+  '@timestamp',
+  'message',
+
+  /* Host */
+  'host.name',
+  'host.ip',
+  'host.os.name',
+  'host.os.version',
+  'host.asset.criticality',
+  'host.risk.calculated_level',
+  'host.risk.calculated_score_norm',
+
+  /* User */
+  'user.name',
+  'user.domain',
+  'user.asset.criticality',
+  'user.risk.calculated_level',
+  'user.risk.calculated_score_norm',
+  'user.target.name',
+
+  /* Service */
+  'service.name',
+  'service.id',
+
+  /* Entity */
+  'entity.id',
+  'entity.name',
+  'entity.type',
+  'entity.sub_type',
+
+  /* Agent */
+  'agent.id',
+
+  /* Process */
+  'process.name',
+  'process.pid',
+  'process.args',
+  'process.command_line',
+  'process.executable',
+  'process.exit_code',
+  'process.working_directory',
+  'process.pe.original_file_name',
+  'process.hash.md5',
+  'process.hash.sha1',
+  'process.hash.sha256',
+  'process.code_signature.exists',
+  'process.code_signature.signing_id',
+  'process.code_signature.status',
+  'process.code_signature.subject_name',
+  'process.code_signature.trusted',
+
+  /* Process parent */
+  'process.parent.name',
+  'process.parent.args',
+  'process.parent.args_count',
+  'process.parent.command_line',
+  'process.parent.executable',
+  'process.parent.code_signature.exists',
+  'process.parent.code_signature.status',
+  'process.parent.code_signature.subject_name',
+  'process.parent.code_signature.trusted',
+
+  /* File */
+  'file.name',
+  'file.path',
+  'file.Ext.original.path',
+  'file.hash.sha256',
+
+  /* Groups */
+  'group.id',
+  'group.name',
+
+  /* Cloud */
+  'cloud.provider',
+  'cloud.account.name',
+  'cloud.service.name',
+  'cloud.region',
+  'cloud.availability_zone',
+
+  /* Network / DNS */
+  'source.ip',
+  'destination.ip',
+  'network.protocol',
+  'dns.question.name',
+  'dns.question.type',
+
+  /* Event */
+  'event.category',
+  'event.action',
+  'event.type',
+  'event.code',
+  'event.dataset',
+  'event.module',
+  'event.outcome',
+
+  /* Rule (generic) */
+  'rule.name',
+  'rule.reference',
+
+  /* Kibana alert fields */
+  'kibana.alert.uuid',
```

Comment:
Just an idea, but we could retrieve some of these from [here](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-rule-data-utils/src/default_alerts_as_data.ts) which is a platform share package?